lib/ssh/test/ssh_agent_SUITE.erl

```diff
@@ -0,0 +1,130 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2019. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+
+%%
+
+-module(ssh_agent_SUITE).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("ssh/src/ssh.hrl").
+-include_lib("ssh/src/ssh_agent.hrl").
+-include("ssh_test_lib.hrl").
+
+-compile(export_all).
+
+%% Test configuration
+
+suite() ->
+    [{timetrap, {seconds, 30}}].
+
+all() ->
+    [request_identities, sign_request, connect_with_ssh_agent].
+
+init_per_suite(Config) ->
+    ok = ssh:start(),
+    ssh_agent_mock_server:check_mktemp(Config).
```

Comment:
We run the tests on 25+ machines with varying architecture, OSes and cryptolib variants.
Some does not even have a cryptolib, so our ssh (and ssl/tls) test suites need to test for crypto availability. If not available, the test suite is skipped. So I suggest:
```suggestion
    ?CHECK_CRYPTO(
       begin
           ok = ssh:start(),
           ssh_agent_mock_server:check_mktemp(Config)
       end
      ).
```
The ?CHECK_CRYPTO will do the begin ... end if crypto is available and skip the suite if it is not available.